src/queue.js

```diff
@@ -0,0 +1,98 @@
+let instance = null
+
+class Queue {
+
+  constructor() {
+    if(!instance){
+      instance = this // eslint-disable-line consistent-this
+      this.setup()
+    }
+    return instance
+  }
+
+  setup() {
+    this.tasks = []
+    this.pendingCounter = {}
+    this.ranCounter = {}
+    this.running = false
+  }
+
+  valid(task) {
+    if(
+      typeof task.run !== 'function' ||
+      typeof task.error !== 'function' ||
+      typeof task.done !== 'function'
+    ) throw new Error('Task has an invalid object signature, methods required')
```

Comment:
Agree with TypeError, but I desagree about returnig false in this case. This method not only serves the purpose but implements the principle of Duck Typing because in Javascript we can't use Interfares so any class that does not implement those methods sholud throw an error because we can't let the implementation throw the error every time this method is being used.